docs/weave/weave-api-spec/03-querying.md

```diff
@@ -0,0 +1,51 @@
+---
+id: weave-query-spec
+title: Querying Weave
+sidebar_label: Querying Weave
+---
+
+In this section querying weave for data will be explained. Before diving into the details, Weave internals such as __buckets__ and __paths__ must be explained.
+
+## Buckets
+
+Buckets are the structures that enable accessing and writing to __Key-Value__ database in an organized and controlled manner. Buckets stores the data as well as the indexes, secondary indexes that refer to the data.
+
+### Accessing buckets
+
+[//]: # (TODO give reference to weave/tendermint or abci documentation)
+As mentioned in the previous sections, Weave uses tendermint as consensus engine thus queries are made to data store via `abci_queries`. Refer to [tendermint/abciquery](https://tendermint.com/rpc/#abciquery).
+
+Via running the JSON-RPC/HTTP call below, __hugnet__ testnet could be queried for retrieving all the tokens registered under `tokens` bucket:
+
+``` bash
+curl -X POST -d '{ "json-rpc": 2.0, "id": "foobar321", "method": "abci_query", "params": { "path": "/tokens?prefix", "data": "" } }' https://bns.hugnet.iov.one/
+```
+
+### Bucket paths
+
+Every bucket registered under Weave is accessed using __paths__. On the example curl command above given path parameter says the bucket wanted to be accessed is the tokens bucket. 
+
+Here are the major bucket paths registered under BNSD:
+
+| Bucket           | Path          | Reference                                                                                                     |
+|------------------|---------------|---------------------------------------------------------------------------------------------------------------|
+| __Cash__             | `wallets`     | [weave/x/cash/handler](https://github.com/iov-one/weave/blob/v0.18.0/x/cash/handler.go#L14)                   |
+| __Sigs__             | `auth`        | [weave/x/sigs/handler](https://github.com/iov-one/weave/blob/v0.18.0/x/sigs/decorator.go#L19)                                |
+| __MultiSig__         | `contracts`   | [weave/x/multisig/handler](https://github.com/iov-one/weave/blob/v0.18.0/x/multisig/handlers.go#L22)          |
+| __Atomic swaps__     | `aswaps`      | [weave/x/aswaps/handler](https://github.com/iov-one/weave/blob/v0.18.0/x/aswap/handler.go#L34)                |
+| __Escrow__           | `escrows`     | [weave/x/escrow/handler](https://github.com/iov-one/weave/blob/v0.18.0/x/escrow/handler.go#L35)               |
```

Comment:
__Escrow__: link to https://github.com/iov-one/weave/blob/v0.18.0/x/escrow/model.go#L113-L117

* `/escrows` -> takes `id` (8 bytes), returns `x/escrow.Escrow` by id
* `/escrows/source` -> takes source `Address` (20 bytes)
* `/escrows/destination` -> takes destination `Address` (20 bytes)
* `/escrows/arbiter` -> takes arbiter `Address` (20 bytes)

Reply:
Linking to the Buckets directly is probably best, so you capture all indexes